PayPal/src/test/java/com/braintreepayments/api/PayPalLauncherUnitTest.kt

```diff
@@ -0,0 +1,124 @@
+package com.braintreepayments.api
+
+import android.content.Intent
+import android.net.Uri
+import androidx.activity.ComponentActivity
+import io.mockk.every
+import io.mockk.mockk
+import org.json.JSONException
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertSame
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class PayPalLauncherUnitTest {
+    private val browserSwitchClient: BrowserSwitchClient = mockk(relaxed = true)
+    private val activity: ComponentActivity = mockk(relaxed = true)
+    private val paymentAuthRequestParams: PayPalPaymentAuthRequestParams = mockk(relaxed = true)
+    private val intent: Intent = mockk(relaxed = true)
+    private val options: BrowserSwitchOptions = mockk(relaxed = true)
+    private val browserSwitchRequest = BrowserSwitchRequest(
+        1,
+        Uri.parse("http://"),
+        JSONObject().put("test_key", "test_value"),
+        "return-url-scheme",
+        false
+    )
+
+    @Before
+    fun setup() {
+        every { paymentAuthRequestParams.browserSwitchOptions } returns options
+    }
+
+    @Test
+    fun launch_startsBrowserSwitch_returnsPendingRequest() {
+        val startedPendingRequest = BrowserSwitchPendingRequest.Started(browserSwitchRequest)
+        every { browserSwitchClient.start(activity, options) } returns startedPendingRequest
+
+        val sut = PayPalLauncher(browserSwitchClient)
```

Comment:
Would we be able to move `sut` to be a lateinit var on the test class, then in `setup` save a new instance of PayPalLauncher? It could minimally clean up each of the test functions.

Reply:
Done 👍